services/venice-controller/src/main/java/com/linkedin/venice/controller/Admin.java

```diff
@@ -935,6 +936,12 @@ default boolean isAdminTopicConsumptionEnabled(String clusterName) {
 
   Map<String, StoreDataAudit> getClusterStaleStores(String clusterName);
 
+  /**
+   * @param clusterName
+   * @return the list of stores ready for compaction
+   */
+  ArrayList<StoreInfo> getStoresForCompaction(String clusterName);
```

Comment:
Replace `ArrayList<Store>` with `List<Store>` as this forces the API to return a concrete implementation of a `List` as opposed to any implementations of `List`; e.g., `LinkedList`, `Vector` etc. 

Generally, using interfaces as return parameters is favorable as it removes the dependencies from callers depending on concrete implementation methods as opposed to well defined publicly exposed methods.

Reply:
done.

I'd like to further my understanding. We want the interface return type to be more generic to reduce dependency on  the return type of the concrete implementation, correct?

What does this mean for the specificity of the return type of the concrete implementation? i.e. should VeniceParentHelixAdmin::getStoresForCompaction() implementation return List or ArrayList?

Reply:
>I'd like to further my understanding. We want the interface return type to be more generic to reduce dependency on the return type of the concrete implementation, correct?

Right. Ideally, you want the methods to return generic interfaces where possible unless you don't have one.

Reply:
I'm starting to understand. Could we walk through examples when we next sync to lock in my understanding?